File: backend/src/main/java/com/lazyspender/backend/model/AccountAccess.java
package com.lazyspender.backend.model;

import java.time.Instant;

import org.springframework.data.annotation.Id;

import com.google.cloud.spring.data.datastore.core.mapping.Entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "accountAccess")
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class AccountAccess {

    @Id
    private String id;

    private String owner;

    private String delegate;

    private AccessRole role;

    private AccessStatus status;

    private Instant createdAt;

    private Instant respondedAt;
}

File: frontend/types/accountAccess.ts
export enum AccessRole {
  COLLABORATOR = 'COLLABORATOR',
  READ = 'READ'
}

export enum AccessStatus {
  PENDING = 'PENDING',
  ACCEPTED = 'ACCEPTED',
  REJECTED = 'REJECTED',
  REVOKED = 'REVOKED'
}

export interface AccountAccess {
  id: string;
  owner: string;
  delegate: string;
  role: AccessRole;
  status: AccessStatus;
  createdAt: string;
  respondedAt?: string;
  ownerName?: string;
  ownerPictureUrl?: string;
}

export interface CreateAccountAccessRequest {
  email: string;
  role: AccessRole;
}
